warehouse/backend_config.go

```diff
@@ -106,6 +105,12 @@ func (s *backendConfigManager) Subscribe(ctx context.Context) <-chan []model.War
 	ch := make(chan []model.Warehouse, 10)
 	s.subscriptions = append(s.subscriptions, ch)
 
+	s.warehousesMu.Lock()
+	if len(s.warehouses) > 0 {
+		ch <- s.warehouses
+	}
+	s.warehousesMu.Unlock()
```

Comment:
@achettyiitr this version was better, now we're holding the lock for the duration of the whole subscribe. I think we should put this code back.

Reply:
The reason, we are adding this is because there is a race condition happening if we are not subscribers and we get backed config.
So, during subscription also, we are sending the last copy of the data.